transform/transformers/project.py

```diff
@@ -0,0 +1,328 @@
+
+                currency_column.append(currency)
+                total_cost_column.append(total_cost)
+            else:
+                currency_column.append(None)
+                total_cost_column.append(None)
+
+        harvested_properties[CURRENCY] = currency_column
+        harvested_properties[TOTAL_COST] = total_cost_column
+
+    @staticmethod
+    def harvest_keywords(df: DataFrame, harvested_properties: dict) -> None:
+        """
+        Temp method until openaire relase keyword fix
+        Harvests keywords from the specified DataFrame columns, converts them from strings to lists,
+        and store them in the output dictionary.
+        """
+        keywords_list = df.select(KEYWORDS).collect()
+
+        keywords_column = []
+
+        for keywords_row in keywords_list:
+            keywords = keywords_row[KEYWORDS] or None
+
+            if keywords:
+                keywords_to_list = [
+                    title_case_funct(keyword.strip())
+                    for keyword in keywords.split(",")
+                    if keyword.strip()
+                ]
+
+                keywords_column.append(keywords_to_list)
+            else:
+                keywords_column.append(None)
+
+        harvested_properties[KEYWORDS] = keywords_column
+
+    @staticmethod
+    def harvest_eosc_score(spark, df: DataFrame, harvested_properties: dict) -> None:
+        """
+        Harvests EOSC score and updates the 'harvested_properties' dictionary.
+
+        Parameters:
+            spark (SparkSession): The Spark session.
+            df (DataFrame): The input DataFrame.
+            harvested_properties (dict): The dictionary to be updated with EOSC score.
+        """
+
+        def create_conditions(joined_dataframe: DataFrame) -> List[when]:
+            """Create a list of conditions for EOSC score calculation."""
+            return [
+                when(
+                    (
+                        ((col(column) != "") & (col(column).isNotNull()))
+                        if joined_dataframe.schema[column].dataType == StringType()
+                        else ((size(col(column)) > 0) & (col(column).isNotNull()))
+                        if joined_dataframe.schema[column].dataType
+                        == ArrayType(StringType())
+                        else col(column).isNotNull()
+                    ),
+                    lit(1),
+                ).otherwise(lit(0))
+                for column in joined_dataframe.columns
+            ]
+
+        def calculate_eosc_score(
+            dataframe: DataFrame, conditions: List[when]
+        ) -> List[int]:
+            """Calculate the EOSC score and return a list of scores."""
+            eosc_score_column = sum(conditions).alias(EOSC_SCORE)
+            dataframe = dataframe.withColumn(EOSC_SCORE, eosc_score_column)
+            return dataframe.select(EOSC_SCORE).rdd.flatMap(lambda x: x).collect()
+
+        harvested_properties_schema = StructType(
```

Comment:
```
        harvested_properties_schema = StructType(
            [
                StructField("currency", StringType(), True),
                StructField("funding_stream_title", ArrayType(StringType()), True),
                StructField("funding_title", ArrayType(StringType()), True),
                StructField("keywords", ArrayType(StringType()), True),
                StructField("total_cost", FloatType(), True),
            ]
        )

        harvested_properties_df = create_df(
            harvested_properties, harvested_properties_schema, spark
        )

        project_df = df.select(
            df[ABBREVIATION],
            df[CODE],
            df[DESCRIPTION],
            df[END_DATE],
            df[OPEN_ACCESS_MANDATE_FOR_DATASET],
            df[OPEN_ACCESS_MANDATE_FOR_PUBLICATIONS],
            df[START_DATE],
            df[SUBJECT],
            df[TITLE],
        )

        joined_df = join_different_dfs((project_df, harvested_properties_df))
```
That should be a part of this method. Its out of its scope and joining Sparks Ddfs is really costly operation. 
Lets have a quick meeting on Monday regarding this issue

Reply:
I added this method as a usability test. If we decide to develop the method, I included a `# TODO REFACTOR`.